flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/RandomAccessInputView.java

```diff
@@ -45,9 +45,12 @@ public RandomAccessInputView(ArrayList<MemorySegment> segments, int segmentSize)
 	{
 		this(segments, segmentSize, segmentSize);
 	}
-	
+
+	/**
+	 * Warning: setReadPosition has to be called before reading.
+	 */
 	public RandomAccessInputView(ArrayList<MemorySegment> segments, int segmentSize, int limitInLastSegment) {
```

Comment:
I tried this, but it turned out that there is no nice way to do it, because the two constructors would have the same parameters, so I had to add a dummy parameter. (See https://github.com/apache/flink/commit/508ce22c3167dc6e6e2a4a6b9776b6c79e62698f) Do you think that this ugly solution is worth it, or should I revert to the original?